src/utils/pluralkit.ts

```diff
@@ -0,0 +1,42 @@
+import {Message, PartialMessage} from 'discord.js';
+import {PersistentData} from '../types/persist';
+
+
+// TODO: When scheduler is implemented make the cache reset after x time
+const cache: string[] = [];
+
+export async function isSystemMessage(message: Message | PartialMessage) {
+	try {
+		// simple fetch to check if the message was proxied
+		return (await (await fetch(`https://api.pluralkit.me/v2/messages/${message.id}`)).text()).includes('Message not found');
+	} catch (e) {
+		return false;
+	}
+}
+
+export async function shouldLog(message: Message | PartialMessage, data: PersistentData): Promise<boolean> {
```

Comment:
should use `data` like a global variable, like every other file that interacts with it

yes its yucky but we don't have a database yet